staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -161,6 +161,10 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("listKind"), names.ListKind, "may have mixed case, but should otherwise match: "+strings.Join(errs, ",")))
 	}
 
+	if strings.HasSuffix(names.Kind, "List") {
```

Comment:
Slightly tangential but some other Kinds should also not be allowed. See https://github.com/kubernetes/kubernetes/pull/59346 (this PR needs a rebase).

I think we should try to factor out all these cases into a single "these kinds are not allowed", rather than restricting each individually. Wdyt?

Reply:
SGTM.